standalone/src/models/shipping-method/zone-rate/builders.spec.ts

```diff
@@ -0,0 +1,96 @@
+import type { TZoneRateRest, TZoneRateGraphql } from './types';
+import { ZoneRateRest, ZoneRateGraphql, ZoneRate } from './index';
+
+function validateRestModel(model: TZoneRateRest) {
+  expect(model).toEqual(
+    expect.objectContaining({
+      zone: expect.objectContaining({
+        typeId: 'zone',
+      }),
+      shippingRates: expect.arrayContaining([
+        expect.objectContaining({
+          freeAbove: null,
+          price: expect.objectContaining({
+            centAmount: expect.any(Number),
+            currencyCode: expect.any(String),
+          }),
+          tiers: [],
+          isMatching: null,
+        }),
+      ]),
+    })
+  );
+}
+
+function validateGraphqlModel(model: TZoneRateGraphql) {
+  const zoneId = model.zone?.id;
+
+  expect(model).toEqual(
```

Comment:
When validating nested objects, it's enough to only check the `__typename` property. If that value is correct, we can assume the object is the one we're expecting.